promises-demo-brightscript/components/MainScene.brs

```diff
@@ -0,0 +1,191 @@
+sub init()
+    initOnScreenImage()
+    m.top.observeFieldScoped("simpleResult", "onSimpleResultChange")
+    m.top.observeFieldScoped("chainResult", "onChainResultChange")
+    m.top.observeFieldScoped("parallelResult", "onParallelResultChange")
+
+    'Uncomment the example below to run...
+
+    'Simple single request
+    ' simpleExample("http://ip-api.com/json/")
+
+    'Chain requests
+    ' chainExample()
+
+    'Parallel requests
+    ' parallelExample()
+
+    'Simple single request with error response
+    ' simpleExample("http://invalid--url.com")
+end sub
+
+function networkRequest(url as string, method = "GET" as string, body = {} as object) as object
+    promise = promises_create()
+    task = createObject("roSGNode", "NetworkTask")
+    task.url = url
+    task.method = method
+    task.body = body
+    task.promise = promise
+    task.control = "RUN"
+    return promise
+end function
+
+sub simpleExample(url as string, method = "GET" as string)
+    context = {
+        result: invalid
+    }
+
+    promise = networkRequest(url, method)
+
+    promises_onThen(promise, sub(response as object, context = {} as dynamic)
+        if (response.error = invalid)
+            context.result = "Your timezone is " + response.timezone
+        else
+            'See: https://developer.roku.com/en-gb/docs/references/brightscript/events/rourlevent.md#getresponsecode-as-integer
+            print "Got a network error response!!!", response.responseCode
+        end if
+
+        m.top.simpleResult = context
+
+    end sub, context)
+
+    promises_onCatch(promise, sub(response as object, context = {} as dynamic)
+        print "Caught a Simple promise error!!!", response
+    end sub, context)
+
+    promises_onFinally(promise, sub(response as object, context = {} as dynamic)
+        print "Simple promise completed!!!"
+    end sub, context)
+end sub
+
+sub chainExample()
+    context = {
+        result: invalid
+    }
+
+    promise = networkRequest("http://ip-api.com/json/")
+
+    promises_chain(promise, context).then(function(response, context)
+        if (response.error = invalid)
```

Comment:
We need to handle what happens in the `else` flow. Because the else will "return" invalid, then the next `.then` block will fail because `response` is invalid.